prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -271,12 +284,21 @@ class ReportFunction(
             .build()
 
         // queue messages here after all task / action records are in
-        actionHistory.queueMessages(workflowEngine)
+        if (!bypassMessageQueue)
+            actionHistory.queueMessages(workflowEngine)
```

Comment:
See if there's a way to not have to check this condition in multiple places.

Reply:
@whytheplatypus i was able to remove one of these conditionals due to the fact that I'm now throwing an ActionError from routeReport() in WorkflowEngine.kt. But I think everything else is needed at this point